plugins/mongo_db/mongo_db_writer.cpp

```diff
@@ -92,10 +95,14 @@ namespace mongo_db {
 
                         state_writer st_writer(all_docs, block);
 
-                        st_writer.write_global_property_object(dgp_s[head_iter->first], head_iter->second, true);
+                        if (store_history_mode_dgp != 0 && (head_iter->second.block_num() % std::abs(store_history_mode_dgp) == 0)) {
+                            st_writer.write_global_property_object(dgp_s[head_iter->first], head_iter->second, true);
+                        }
                         st_writer.write_global_property_object(dgp_s[head_iter->first], head_iter->second, false);
 
-                        st_writer.write_witness_schedule_object(wso_s[head_iter->first], head_iter->second, true);
+                        if (store_history_mode_wso != 0 && (head_iter->second.block_num() % std::abs(store_history_mode_wso) == 0)) {
```

Comment:
`std::abs()` is some magic for users. They will try to set different numbers (negative and positive) and try to understand how it works.